pkg/occlient/occlient.go

```diff
@@ -533,16 +541,50 @@ func (c *Client) CreateNewProject(projectName string, wait bool) error {
 			if !ok {
 				break
 			}
-			if e, ok := val.Object.(*projectv1.Project); ok {
-				glog.V(4).Infof("Project %s now exists", e.Name)
-				return nil
+			if prj, ok := val.Object.(*projectv1.Project); ok {
+				glog.V(4).Infof("Status of creation of project %s is %s", prj.Name, prj.Status.Phase)
+				switch prj.Status.Phase {
+				//prj.Status.Phase can only be "Terminating" or "Active" or ""
+				case corev1.NamespaceActive:
+					if val.Type == watch.Added {
+						glog.V(4).Infof("Project %s now exists", prj.Name)
+						return nil
+					}
+					if val.Type == watch.Error {
+						return fmt.Errorf("failed watching the deletion of project %s", prj.Name)
+					}
+				}
 			}
 		}
 	}
 
 	return nil
 }
 
+func (c *Client) WaitForDefaultServiceAccountInNamespace(namespace, serviceAccountName string) error {
```

Comment:
The function name is a little bit confusing.  It is not waiting for default service account, but for the service account specified by `serviceAccountName` argument.

You can just call it `WaitForServiceAccount`. 
There should also be a  comment describing what is this function doing. like for example that it will block